.github/workflows/check-projects.yml

```
@@ -0,0 +1,99 @@
+name: check-projects
+on:
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  check-matsim-libs:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout PR
+        uses: actions/checkout@v2
+        with:
+          persist-credentials: false
+          
+      - name: Checkout Baseline
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.pull_request.base.ref }}
+          path: spotbugs-baseline
+          persist-credentials: false
+
+      - name: Checkout Matsim-libs
+        uses: actions/checkout@v2
+        with:
+          repository: 'matsim-org/matsim-libs'
+          ref: 'matsim-12.0'
+          path: matsim-libs
+          persist-credentials: false
+
+      - name: Set up JDK 15
+        uses: actions/setup-java@v1
+        with:
+          java-version: '15.0.x'
+
+      - name: Cache Gradle
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.gradle/caches
+            ~/.gradle/wrapper
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-
```

Java 15 is not supported, better to use Java 17 instead.

```suggestion
      - name: Set up JDK 17
        uses: actions/setup-java@v3
        with:
          java-version: '17.0.x'
          cache: 'gradle'
          distribution: 'temurin'
```